clik/external/hal_cpu/include/cpu_hal.h

```diff
@@ -41,7 +44,9 @@ struct cpu_barrier {
 class cpu_hal : public hal::hal_device_t {
  public:
   cpu_hal(hal::hal_device_info_t *info, std::mutex &hal_lock);
-  virtual ~cpu_hal();
+#if HAL_CPU_MODE == HAL_CPU_WI_MODE
+  ~cpu_hal();
```

Comment:
I could be misremembering but I think we do need to either keep in the `virtual` (for both modes) or mark `cpu_hal` `final`, as otherwise some compilers will warn.